src/views/CountryRiskWatch/RiskBarChart/FoodInsecurityChart/index.tsx

```diff
@@ -0,0 +1,294 @@
+    Fragment,
+    useCallback,
+    useMemo,
+    useRef,
+} from 'react';
+import {
+    isDefined,
+    isNotDefined,
+    listToGroupList,
+    listToMap,
+    mapToMap,
+} from '@togglecorp/fujs';
+
+import useSizeTracking from '#hooks/useSizeTracking';
+import { formatNumber, maxSafe } from '#utils/common';
+import {
+    getAverageIpcData,
+    getPrioritizedIpcData,
+    monthNumberToNameMap,
+} from '#utils/risk';
+import { getDiscretePathDataList, getScaleFunction } from '#utils/chart';
+import {
+    COLOR_HAZARD_FOOD_INSECURITY,
+    COLOR_PRIMARY_RED,
+} from '#utils/constants';
+import { paths } from '#generated/riskTypes';
+
+import ChartAxes from '../ChartAxes';
+import styles from './styles.module.css';
+
+type GetCountryRisk = paths['/api/v1/country-seasonal/']['get'];
+type CountryRiskResponse = GetCountryRisk['responses']['200']['content']['application/json'];
+type RiskData = CountryRiskResponse[number];
+
+const colors = [
+    '#cbcedb',
+    '#a9adbf',
+    '#878ea5',
+    '#676f8a',
+    '#475271',
+    '#283759',
+    '#011e41',
+];
+
+const X_AXIS_HEIGHT = 16;
+const Y_AXIS_WIDTH = 32;
+const CHART_OFFSET = 16;
+
+const chartMargin = {
+    left: Y_AXIS_WIDTH + CHART_OFFSET,
+    top: CHART_OFFSET,
+    right: CHART_OFFSET,
+    bottom: X_AXIS_HEIGHT + CHART_OFFSET,
+};
+
+const xAxisFormatter = (date: Date) => date.toLocaleString(
+    navigator.language,
+    { month: 'short' },
+);
+
+const formatDate = (date: Date) => date.toLocaleString(
+    navigator.language,
+    {
+        year: 'numeric',
+        month: 'short',
+    },
+);
+
+interface Props {
+    ipcData: RiskData['ipc_displacement_data'] | undefined;
+}
+
+function FoodInsecurityChart(props: Props) {
+    const { ipcData } = props;
+
+    const chartContainerRef = useRef<HTMLDivElement>(null);
+    const chartBounds = useSizeTracking(chartContainerRef);
+
+    const chartData = useMemo(
+        () => {
+            const uniqueData = getPrioritizedIpcData(ipcData ?? []);
+            const averageDataItem = getAverageIpcData(uniqueData);
+            const maxValue = maxSafe(uniqueData.map((item) => item.total_displacement)) ?? 0;
+
+            const xScale = getScaleFunction(
+                { min: 0, max: 12 },
+                { min: 0, max: chartBounds.width },
+                { start: chartMargin.left, end: chartMargin.right },
+            );
+
+            const yScale = getScaleFunction(
+                { min: 0, max: maxValue },
+                { min: 0, max: chartBounds.height },
+                { start: chartMargin.top, end: chartMargin.bottom },
+                true,
+            );
+
+            const yearGroupedData = mapToMap(
+                listToGroupList(
+                    uniqueData,
+                    (item) => item.year,
+                ),
+                (year) => year,
+                (ipcDataList) => getAverageIpcData(ipcDataList),
+            );
+
+            const yearKeys = Object.keys(yearGroupedData);
```

Comment:
What if the yearGroupedData is empty?

Reply:
We'll get empty list for `yearKeys`

Reply:
We should avoid indexing with `-1`.
Check line 110